### PR TITLE
Replace SubqueryExpr with logical node

### PR DESCRIPTION
**Pull request by fpetkovski**
This commit replaces the SubqueryExpr with a custom logical node.

### CODE REVIEW

File: logicalplan/options.go [hunk 1]
```diff
@@ -0,0 +1,4 @@
+// Copyright (c) The Thanos Community Authors.
+// Licensed under the Apache License 2.0.
+
+package logicalplan
```
Comment (MichaHoffmann) [inline logicalplan/options.go:4]: do we need this file still?

File: logicalplan/plan.go [hunk 2]
```diff
@@ -247,7 +247,13 @@ func replacePrometheusNodes(plan parser.Expr) parser.Expr {
 	case *parser.StepInvariantExpr:
 		return &StepInvariantExpr{Expr: replacePrometheusNodes(t.Expr)}
 	case *parser.MatrixSelector:
-		return &MatrixSelector{VectorSelector: replacePrometheusNodes(t.VectorSelector), Range: t.Range, OriginalString: t.String()}
+		return &MatrixSelector{
+			VectorSelector: &VectorSelector{
+				VectorSelector: t.VectorSelector.(*parser.VectorSelector),
```
Comment (MichaHoffmann) [inline logicalplan/plan.go:252]: did recursion break?

**Review by MichaHoffmann — APPROVED** [review]
lgtm, small nit

File: logicalplan/plan.go (hunk 2 above)
Comment (fpetkovski) [reply logicalplan/plan.go:252]: I made `VectorSelector` a concrete type in MarixSelector so we would avoid casts in the rest of the codebase.

Comment (MichaHoffmann) [reply logicalplan/plan.go:252]: Ah! Yeah makes sense!

**Review by MichaHoffmann — APPROVED** [review]
Lgtm

**Pull request merged**